packages/mailchimp/src/Utils.js

```diff
@@ -1,4 +1,21 @@
 import { composeNextState } from '@openfn/language-common';
+import { Client, MockAgent } from 'undici';
+
+const clients = new Map();
```

Comment:
Good - but this is actually a bit over-complicated for mailchimp! There is only ever one `baseUrl` in the config so there's no need to store and maintain a map..

You should be able to do this (and your unit tests should continue to pass)
```
let client;

export const getClient = baseUrl => {
    if (client) {
       return client;
    }
    return  new Client(baseUrl);
}

export const enableMockClient = baseUrl => {
  const mockAgent = new MockAgent({ connections: 1 });
  client = mockAgent.get(baseUrl);
  return client;
};
```

It is still a slightly strange pattern and I'm going to have to think about it a bit more in http land. But it's fine for here today

Reply:
✅ [62ced82](https://github.com/OpenFn/adaptors/pull/352/commits/62ced82eb39655501c2cde9e747c39edfcf9daff)